src/jit/codegenxarch.cpp

```diff
@@ -2768,7 +2778,8 @@ void CodeGen::genCodeForInitBlkRepStos(GenTreeBlk* initBlkNode)
 #ifdef _TARGET_AMD64_
         assert(size > CPBLK_UNROLL_LIMIT && size < CPBLK_MOVS_LIMIT);
 #else
-        assert(size > CPBLK_UNROLL_LIMIT);
+        // Note that a size of zero means a non-constant size.
+        assert((size == 0) || (size > CPBLK_UNROLL_LIMIT));
```

Comment:
Does this mean that the earlier optimization phases are required to remove any such InitBlk/CopyBlks which actually have a constant zero size?  If so we need tests for such cases.  And via copy prop as well we could have a InitBlk/CopyBlk with a LclVar as the size and then via constant prop we would replace the LclVar with a constant zero.